programs/psinode/main.cpp

```diff
@@ -1091,9 +1095,84 @@ struct TLSConfig
 };
 PSIO_REFLECT(TLSConfig, certificate, key, trustfiles)
 
+struct ExtraConfigField
+{
+   std::variant<std::string, std::vector<std::string>> value;
+};
+
+void to_json(const ExtraConfigField& field, auto& stream)
+{
+   std::visit([&](auto& v) { to_json(v, stream); }, field.value);
+}
+
+void from_json(ExtraConfigField& field, auto& stream)
+{
+   auto t = stream.peek_token();
+   switch (t.get().type)
+   {
+      case psio::json_token_type::type_bool:
+         field.value = stream.get_bool() ? "true" : "false";
+         break;
+      case psio::json_token_type::type_string:
+         field.value = std::string(stream.get_string());
+         break;
+      case psio::json_token_type::type_start_array:
+      {
+         std::vector<std::string> values;
+         while (!stream.get_end_array_pred())
```

Comment:
Is it possible this isn't consuming the start of the array token correctly? I'm just seeing peeks...